tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ReceiveMessageIntegration.cs

```diff
@@ -68,7 +72,37 @@ public class ReceiveMessageIntegration
         /// <param name="state">Calltarget state value</param>
         /// <returns>A response value, in an async scenario will be T of Task of T</returns>
         internal static CallTargetReturn<TResponse> OnMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception exception, in CallTargetState state)
+            where TResponse : IReceiveMessageResponse
         {
+            if (response.Instance != null && response.Messages.Count > 0)
+            {
+                var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+                if (dataStreamsManager != null && dataStreamsManager.IsEnabled)
+                {
+                    var queueName = (string)state.State;
```

Comment:
This _could_ be null. How should we handle that e.g. in the edge tags?

```suggestion
                    var queueName = (string?)state.State;
```